samples/apps/copilot-chat-app/webapp/src/App.tsx

```diff
@@ -69,26 +69,38 @@ const App: FC = () => {
     const dispatch = useAppDispatch();
 
     const { instance, inProgress } = useMsal();
-    const account = instance.getActiveAccount();
+    const { loggedInUserInfo } = useAppSelector((state: RootState) => state.app);
     const isAuthenticated = useIsAuthenticated();
 
     const chat = useChat();
 
     useEffect(() => {
-        if (isAuthenticated && account) {
-            dispatch(setLoggedInUserId(account.homeAccountId));
-
-            if (appState === AppState.LoadingChats) {
-                // Load all chats from the backend.
-                async function loadChats() {
-                    if (await chat.loadChats()) {
-                        setAppState(AppState.Chat);
-                    }
+        if (isAuthenticated) {
+            if (loggedInUserInfo === undefined) {
+                const account = instance.getActiveAccount();
+                if (!account) {
+                    dispatch(addAlert({ type: AlertType.Error, message: 'Unable to get active logged in account.' }));
+                } else {
+                    dispatch(setLoggedInUserInfo({
+                        id: account.homeAccountId,
+                        email: account.username,
+                        fullName: account.name ?? account.username,
+                    }));
                 }
+            } else {
+                if (appState === AppState.LoadingChats) {
```

Comment:
nit: `else if`

Reply:
the if statement should be embedded in an else {} block because we only want to load the chats if the LoggedInUserInfo has been defined since we have dependencies on the user id

Reply:
since the `if` statement is the only logic in the `else` block, the `else if` will be equivalent

Reply:
Actually we should get rid of the else block because we want to carry on loading the chats after the user info is set.